aria_data_loaders/aria_data_utils/spot_aria_p1_node.py

```diff
@@ -0,0 +1,591 @@
+from bosdyn.client.frame_helpers import get_a_tform_b
+from fairotag.scene import Scene
+from geometry_msgs.msg import Pose, PoseStamped
+from matplotlib import pyplot as plt
+from projectaria_tools.core import calibration, data_provider, mps
+from scipy.spatial.transform import Rotation as R
+from spot_rl.envs.skill_manager import SpotSkillManager
+from spot_rl.models.owlvit import OwlVit
+from spot_wrapper.spot import Spot, SpotCamIds, image_response_to_cv2
+from std_msgs.msg import Bool
+from tf2_ros import (
+    ConnectivityException,
+    ExtrapolationException,
+    LookupException,
+    StaticTransformBroadcaster,
+)
+from tf2_ros.buffer import Buffer
+from tf2_ros.transform_listener import TransformListener
+
+DOCK_ID = int(os.environ.get("SPOT_DOCK_ID", 520))
+FILTER_DIST = 2.4  # in meters (distance for valid detection)
+
+
+class SpotQRDetector:
+    def __init__(self, spot: Spot):
+        self.spot = spot
+        print("...Spot initialized...")
+
+    def _to_camera_metadata_dict(self, camera_intrinsics):
+        """Converts a camera intrinsics proto to a 3x3 matrix as np.array"""
+        intrinsics = {
+            "fx": camera_intrinsics.focal_length.x,
+            "fy": camera_intrinsics.focal_length.x,
+            "ppx": camera_intrinsics.principal_point.x,
+            "ppy": camera_intrinsics.principal_point.y,
+            "coeffs": [0.0, 0.0, 0.0, 0.0, 0.0],
+        }
+        return intrinsics
+
+    def _get_body_T_handcam(self, frame_tree_snapshot_hand):
+        hand_bd_wrist_T_handcam_dict = (
+            frame_tree_snapshot_hand.child_to_parent_edge_map.get(
+                "hand_color_image_sensor"
+            ).parent_tform_child
+        )
+        hand_mn_wrist_T_handcam = self.spot.convert_transformation_from_BD_to_magnum(
+            hand_bd_wrist_T_handcam_dict
+        )
+
+        hand_bd_body_T_wrist_dict = (
+            frame_tree_snapshot_hand.child_to_parent_edge_map.get(
+                "arm0.link_wr1"
+            ).parent_tform_child
+        )
+        hand_mn_body_T_wrist = self.spot.convert_transformation_from_BD_to_magnum(
+            hand_bd_body_T_wrist_dict
+        )
+
+        hand_mn_body_T_handcam = hand_mn_body_T_wrist @ hand_mn_wrist_T_handcam
+
+        return hand_mn_body_T_handcam
+
+    def get_spot_a_T_b(self, a: str, b: str) -> sp.SE3:
+        frame_tree_snapshot = (
+            self.spot.get_robot_state().kinematic_state.transforms_snapshot
+        )
+        se3_pose = get_a_tform_b(frame_tree_snapshot, a, b)
+        pos = se3_pose.get_translation()
+        quat = se3_pose.rotation.normalize()
+        return sp.SE3(quat.to_matrix(), pos)
+
+    def _get_body_T_headcam(self, frame_tree_snapshot_head):
+        raise RuntimeError("This method is not used anymore. Please use get_spot_a_T_b")
+
+    def _get_spotWorld_T_handcam(
+        self, frame_tree_snapshot_hand, spot_frame: str = "vision"
+    ):
+        if spot_frame != "vision" and spot_frame != "odom":
```

Comment:
Instead of "vision" or "odom" these should probably be constants being accessed as part of Spot class?

Reply:
added